### PR TITLE
Updates for Top and BTV

### PR DESCRIPTION
**Pull request by defranchis**
Based on https://github.com/cms-sw/cmssw/pull/19119

### CODE REVIEW

**Comment by mtosi** [comment]
where is HLT_Mu23_TrkIsoVVL_Ele8_CaloIdL_TrackIdL_IsoVL_v* ?
it appears in the google doc, but I do not see in the monitoring :(

**Comment by defranchis** [comment]
@mtosi there must be some kind of mistake since we are not using HLT_Mu23_TrkIsoVVL_Ele8_CaloIdL_TrackIdL_IsoVL_v*

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
@defranchis this PR has automatically been moved to CMSSW 9.3.x .
Can you make a new PR for CMSSW 9.2.x ?
I think at the moment you can use the same branch.

**Comment by defranchis** [comment]
@fwyzard here it is https://github.com/cms-sw/cmssw/pull/19505, both for this PR and for https://github.com/cms-sw/cmssw/pull/19119

**Comment by defranchis** [comment]
@fwyzard here it is https://github.com/cms-sw/cmssw/pull/19505

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
> @fwyzard here it is #19505

Thank you.
.A

**Comment by davidlange6** [comment]
@defranchis  while waiting for your DQM review - there is now a base class appropriate for your new modules available at https://github.com/cms-sw/cmssw/blob/master/DQMOffline/Trigger/plugins/TriggerDQMBase.h. Please have a look and adapt your module. In addition to removing code duplication, you will have addressed a number of the "usual" comments on PRs like this one likely resulting in a faster review.

**Comment by dmitrijus** [comment]
+1
(This is already merged upstream,  the base module can/should be adapted later on)

**Comment by mtosi** [comment]
please, rebase

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by defranchis** [comment]
@mtosi can we please test?

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by dmitrijus** [comment]
+1

File: DQMOffline/Trigger/plugins/TopMonitor.cc [hunk 1]
```diff
@@ -639,6 +641,27 @@ void TopMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup
 
   if (bjets.size() < nbjets_ ) return;  
 
+  if (nbjets_ > 1){
+      double deltaEta = abs(bjets.begin()->first->eta()-(++bjets.begin())->first->eta());
```
Comment (davidlange6) [inline DQMOffline/Trigger/plugins/TopMonitor.cc]: hi @defranchis - please use std::abs

File: DQMOffline/Trigger/plugins/TopMonitor.cc [hunk 2]
```diff
@@ -639,6 +641,27 @@ void TopMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup
...
+      if (deltaEta > bJetDeltaEtaMax_) return;
+  }
+
+  if ( (nbjets_>0) && (nmuons_>0)){
+      bool foundMuonInsideJet = false;
+      for (auto & bjet : bjets){
+          for (auto & mu : muons){
```
Comment (davidlange6) [inline DQMOffline/Trigger/plugins/TopMonitor.cc]: these should both be const auto &

**Comment by defranchis** [comment]
@davidlange6 let me open a new PR for my convenience

**Comment by defranchis** [comment]
I am closing this PR in favour of https://github.com/cms-sw/cmssw/pull/19641

**Pull request closed without merging**